src/renderer/components/helpers/ChatMethods.ts

```diff
@@ -206,6 +206,49 @@ export async function joinCall(
   }
 }
 
+/**
+ * Creates a new chat with given email address and returns chat id.
+ *
+ * In case an unknown email address was chosen or a chat does not exist yet the
+ * user will be prompted with a confirmation dialogue. In case the user aborts the
+ * action `null` is returned.
+ */
+export async function createChatByEmail(email: string): Promise<number | null> {
+  const accountId = selectedAccountId()
+
+  let contactId = await BackendRemote.rpc.lookupContactIdByAddr(
+    accountId,
+    email
+  )
+
+  const chatId = contactId
+    ? await BackendRemote.rpc.getChatIdByContactId(accountId, contactId)
+    : null
+
+  if (chatId) {
+    return chatId
+  }
+
+  // Ask user if they want to proceed with creating a new contact and / or chat
+  const continueProcess = await new Promise((resolve, _reject) => {
+    window.__openDialog('ConfirmationDialog', {
```

Comment:
optional: tip: if you import the dialog and use it instead of the id here you also get type validation on the options.